java/core/src/java/org/apache/orc/impl/ReaderImpl.java

```diff
@@ -516,12 +516,13 @@ OrcTail buildEmptyTail() {
 
   protected OrcTail extractFileTail(FileSystem fs, Path path,
       long maxFileLength) throws IOException {
-    FSDataInputStream file = fs.open(path);
+    FSDataInputStream file = null;
     ByteBuffer buffer;
     OrcProto.PostScript ps;
     OrcProto.FileTail.Builder fileTailBuilder = OrcProto.FileTail.newBuilder();
     long modificationTime;
     try {
```

Comment:
Maybe replace with try-with-resource block?

Reply:
Thank you for review, @prasanthj .
Ya. I thought to use `try-with-resource` first, but it needs more change to print out the log at [Line 610](https://github.com/apache/orc/pull/218/files#diff-90ceebe6c0bfa3e98b4536455a60b498R610).

Reply:
I think this is the minimal change while keeping the current behavior.

Reply:
Line 610 doesn't add much value as close() will throw IOException anyway which should be sufficient to know the reason.

Reply:
So, you mean it's okay to remove the following in this PR, right?
```
catch (IOException ex) {
  LOG.error("Failed to close the file after another error", ex);
}
```

Reply:
Yeah that's right.